multimixer-128/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+use core::fmt;
+
+use digest::crypto_common::{InvalidLength, Key, KeyInit, KeySizeUser};
+pub use digest::{self, Digest};
+
+use digest::typenum::Unsigned;
+use digest::{
+    consts::{U32, U64},
+    core_api::{
+        AlgorithmName, Block, BlockSizeUser, BufferKindUser, CoreWrapper, FixedOutputCore,
+        UpdateCore,
+    },
+    HashMarker, OutputSizeUser,
+};
+use rand_chacha::rand_core::{RngCore, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+const BLOCKSIZE: usize = 32;
+
+#[derive(Clone)]
+pub struct MultimixerCore {
+    key_blocks: Vec<Block<Self>>,
+    block_sums: [u64; 8usize],
+    block_index: usize,
+    rng: Option<ChaCha8Rng>,
```

Comment:
This looks... suspicious. If you want to make code generic over key blocks generator, I think it's better to make it generic over closure.